java/src/main/java/com/microsoft/edge/seleniumtools/EdgeOptions.java

```diff
@@ -0,0 +1,317 @@
+     *
+     * @param paths Paths to the extensions to install.
+     */
+    public EdgeOptions addExtensions(List<File> paths) {
+        for (File path : paths) {
+            checkNotNull(path);
+            checkArgument(path.exists(), "%s does not exist", path.getAbsolutePath());
+            checkArgument(!path.isDirectory(), "%s is a directory",
+                    path.getAbsolutePath());
+        }
+        extensionFiles.addAll(paths);
+        return this;
+    }
+
+    /**
+     * @param encoded Base64 encoded data of the extensions to install.
+     * @see #addEncodedExtensions(java.util.List)
+     */
+    public EdgeOptions addEncodedExtensions(String... encoded) {
+        addEncodedExtensions(ImmutableList.copyOf(encoded));
+        return this;
+    }
+
+    /**
+     * Adds a new Edge extension to install on browser startup. Each string data should
+     * specify a Base64 encoded string of packed Edge extension (CRX file).
+     *
+     * @param encoded Base64 encoded data of the extensions to install.
+     */
+    public EdgeOptions addEncodedExtensions(List<String> encoded) {
+        for (String extension : encoded) {
+            checkNotNull(extension);
+        }
+        extensions.addAll(encoded);
+        return this;
+    }
+
+    /**
+     * Sets an experimental option. Useful for new EdgeDriver options not yet
+     * exposed through the {@link EdgeOptions} API.
+     *
+     * @param name Name of the experimental option.
+     * @param value Value of the experimental option, which must be convertible
+     *     to JSON.
+     */
+    public EdgeOptions setExperimentalOption(String name, Object value) {
+        experimentalOptions.put(checkNotNull(name), value);
+        return this;
+    }
+
+    /**
+     * Returns the value of an experimental option.
+     *
+     * @param name The option name.
+     * @return The option value, or {@code null} if not set.
+     * @deprecated Getters are not needed in browser Options classes.
+     */
+    @Deprecated
+    public Object getExperimentalOption(String name) {
+        return experimentalOptions.get(checkNotNull(name));
+    }
+
+    public EdgeOptions setPageLoadStrategy(PageLoadStrategy strategy) {
+        setCapability(PAGE_LOAD_STRATEGY, strategy);
+        return this;
+    }
+
+    public EdgeOptions setUnhandledPromptBehaviour(UnexpectedAlertBehaviour behaviour) {
+        setCapability(UNHANDLED_PROMPT_BEHAVIOUR, behaviour);
+        setCapability(UNEXPECTED_ALERT_BEHAVIOUR, behaviour);
+        return this;
+    }
+
+    /**
+     * Returns EdgeOptions with the capability ACCEPT_INSECURE_CERTS set.
+     * @param acceptInsecureCerts
+     * @return EdgeOptions
+     */
+    public EdgeOptions setAcceptInsecureCerts(boolean acceptInsecureCerts) {
+        setCapability(ACCEPT_INSECURE_CERTS, acceptInsecureCerts);
+        return this;
+    }
+
+    public EdgeOptions setHeadless(boolean headless) {
+        args.remove("--headless");

```

Comment:
should we remove --disable-gpu when headless is false?

Reply:
This matches the existing ChromeOptions behavior in Selenium 3. The goal is to provide matching behavior.

I checked if this bug is still present in the latest official Selenium so we can fix it there. It looks like the --disable-gpu option was removed in Selenium 4.0 Alpha 1 so this is no longer a problem: https://github.com/SeleniumHQ/selenium/blob/941dc9c6b2e2aa4f701c1b72be8de03d4b7e996a/py/CHANGES#L87